### ISSUE TITLE
Micromamba and Mamba create conflicting mamba.sh files

### ISSUE THREAD

**Issue by maresb** [mamba-org/mamba#1322]
The directory `$MAMBA_ROOT_PREFIX/etc/profile.d/` contains various scripts which are used during shell activation.

Micromamba writes [this standalone script](https://github.com/mamba-org/mamba/blob/bab7ae50d75f067c493b8d0cd065ea29bd3f9e8c/libmamba/data/mamba.sh) as `mamba.sh`. This script defines the Bash shell function `micromamba()`.

Similarly, Mamba writes [this different script](https://github.com/mamba-org/mamba/blob/bab7ae50d75f067c493b8d0cd065ea29bd3f9e8c/mamba/mamba/shell_templates/mamba.sh) as `mamba.sh`. This script defines `mamba()` and depends on `conda.sh` having been sourced beforehand (it invokes `__conda_activate()`).

Having two scripts with the exact same filename but disjoint functionality leads to errors.

For example, if `mamba.sh` exists, then `micromamba shell hook --shell=bash` uses it as the init script. However, because Mamba's `mamba.sh` doesn't define `micromamba()`, you can't subsequently activate with `micromamba activate xyz`. Doing so results in
```text
The following arguments were not expected: xyz activate
Run with --help for more information.
```
(see also #1323)

This issue is the direct cause of issues #1358 and https://github.com/mamba-org/micromamba-docker/issues/57

**Comment by maresb** [comment]
This problem is even worse than I originally realized. The command `mamba activate xyz` doesn't even work. It fails with

```
__conda_activate: command not found
```

due to `mamba.sh` containing
 
```bash
case "$cmd" in
    activate|deactivate)
        __conda_activate "$cmd" "$@"
        ;;
```

**Comment by maresb** [comment]
As promised, here is the MWE.

Dockerfile:
```Dockerfile
FROM mambaorg/micromamba:git-ab5a635

RUN micromamba install -c conda-forge -y --prefix=$MAMBA_ROOT_PREFIX mamba

ENV MAMBA_SKIP_ACTIVATE=1
```
Host commands:
```bash
docker build -t mamba-1322 .
docker run --rm -it mamba-1322
```

Container commands:
```bash
# Look at the init output:
/bin/micromamba shell hook -s bash
# (The output is $MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh plus shell completions.)

# Initialize the shell
eval "$(/bin/micromamba shell hook -s bash)"

# conda() is undefined because we don't read conda.sh
conda activate base
# bash: conda: command not found

# mamba() is broken since it depends on __conda_activate() from conda.sh
mamba activate base
# bash: __conda_activate: command not found

# micromamba() is undefined
micromamba activate base
# The following arguments were not expected: base activate
# Run with --help for more information.
```

I have discovered an ugly but effective workaround. Micromamba's shell hook will define `micromamba()` if it can't find `$MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh`. Thus by temporarily setting `MAMBA_ROOT_PREFIX` to something invalid, I can trick it into initializing `micromamba()`.

```bash
eval "$(MAMBA_ROOT_PREFIX=/_invalid /bin/micromamba shell hook -s bash)"
micromamba activate base
```

**Comment by maresb** [comment]
I think the proper behavior for the Bash shell hook is:

* Add `$MAMBA_ROOT_PREFIX/etc/profile.d/conda.sh` if it exists.
* Add `$MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh` if it exists.
* Add micromamba hooks and shell completion.

The [current logic](https://github.com/mamba-org/mamba/blob/bab7ae50d75f067c493b8d0cd065ea29bd3f9e8c/libmamba/src/core/activation.cpp#L684-L691) is:

* [Add](https://github.com/mamba-org/mamba/blob/bab7ae50d75f067c493b8d0cd065ea29bd3f9e8c/libmamba/src/core/activation.cpp#L832) `$MAMBA_ROOT_PREFIX/etc/profile.d/mamba.sh` if it exists.
* Otherwise add [micromamba hooks and shell completion](https://github.com/mamba-org/mamba/blob/bab7ae50d75f067c493b8d0cd065ea29bd3f9e8c/libmamba/data/mamba.sh).

I can attempt a PR, but it might turn out funny since I don't know C++ so well.

**Comment by maresb** [comment]
After understanding the code better, I think the proper behavior is:

* Print `$MAMBA_ROOT_PREFIX/etc/profile.d/micromamba.sh` if it exists.
* Otherwise print [micromamba hooks and shell completion](https://github.com/mamba-org/mamba/blob/bab7ae50d75f067c493b8d0cd065ea29bd3f9e8c/libmamba/data/mamba.sh).

I implemented this as the solution in #1355.

**Comment by maresb** [comment]
@wolfv, it has taken me a while to understand exactly why and how everything is failing here. I believe I now understand what's going wrong, have proposed a solution #1355, and I've edited all the descriptions so that hopefully they're complete and coherent.

I'd really appreciate your feedback if you get the chance to take a look.
